docs/dialect_agnostic_sharding.md

```diff
@@ -0,0 +1,94 @@
+# Dialect-agnostic Sharding
+
+The long term goal is to make Shardy a completely standalone component, able to work with any MLIR dialect. Currently,
+Shardy directly depends on StableHLO, but we are making progress towards lifting that through various
+abstractions and interfaces to make Shardy more flexible.
+
+## Sharding Rules
+
+A _sharding rule_ encodes how we propagate through an operation. Since Shardy now depends on StableHLO, it defines
+sharding rules for each stablehlo op. In addition, Shardy provides the [`ShardingRuleOpInterface`](./sdy_op_interfaces#shardingruleopinterface_shardingruleopinterface)
+which can be used by dialect owners in their operations to define sharding rules for their own operations. As long as an
+operation implements this interface, Shardy will be able to propagate through it.
+
+```mlir
+def ShardingRuleOpInterface : OpInterface<"ShardingRuleOpInterface"> {
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/[{
+        Returns the sharding rule of the op.
+      }],
+      /*retType=*/"mlir::sdy::OpShardingRuleAttr",
+      /*methodName=*/"getShardingRule"
+    >,
+  ];
+}
+```
+
+## Region-based ops (Data flow ops)
+
+Region-based ops require a different approach where sharding rules, which only describe the correspondence between
+operands and results, are not enough. In these cases, we need to describe how the sharding is propagated through the
+regions of the op.
+
+Shardy defines a [`ShardableDataFlowOpInterface`](./sdy_op_interfaces#shardabledataflowopinterface_shardabledataflowopinterface)
+so that dialect owners can describe the propagation of sharding through the regions of their ops. This interface provides
```

Comment:
Lets follow what we wrote in the propagation docs:

This interface provides methods to get the sources and targets of each data flow edge through their *owner*, and also get and set the shardings of edge *owners*

Reply:
The current sentence is missing the first part, should be: methods to get the sources and targets of each data flow edge through their owner, and also get and set the shardings of edge owners

Reply:
This hasn't been addressed yet